src/seedu/addressbook/commands/AggregateCommand.java

```diff
@@ -0,0 +1,106 @@
+package seedu.addressbook.commands;
+
+public class AggregateCommand {
+    private AddCommand addCommand;
+    private ClearCommand clearCommand;
+    private DeleteCommand deleteCommand;
+    private ExitCommand exitCommand;
+    private FindCommand findCommand;
+    private HelpCommand helpCommand;
+    private IncorrectCommand incorrectCommand;
+    private ListCommand listCommand;
+    private ViewAllCommand viewAllCommand;
+    private ViewCommand viewCommand;
+
+    public AggregateCommand(AddCommand addCommand) {
+        this.addCommand = addCommand;
+    }
+
+    public AggregateCommand(ClearCommand clearCommand) {
+        this.clearCommand = clearCommand;
+    }
+
+    public AggregateCommand(DeleteCommand deleteCommand) {
+        this.deleteCommand = deleteCommand;
+    }
+
+    public AggregateCommand(ExitCommand exitCommand) {
+        this.exitCommand = exitCommand;
+    }
+
+    public AggregateCommand(FindCommand findCommand) {
+        this.findCommand = findCommand;
+    }
+
+    public AggregateCommand(HelpCommand helpCommand) {
+        this.helpCommand = helpCommand;
+    }
+
+    public AggregateCommand(IncorrectCommand incorrectCommand) {
+        this.incorrectCommand = incorrectCommand;
+    }
+
+    public AggregateCommand(ListCommand listCommand) {
+        this.listCommand = listCommand;
+    }
+
+    public AggregateCommand(ViewAllCommand viewAllCommand) {
+        this.viewAllCommand = viewAllCommand;
+    }
+
+    public AggregateCommand(ViewCommand viewCommand) {
+        this.viewCommand = viewCommand;
+    }
+
+    public CommandResult execute() {
+        if (addCommand != null) {
+            return addCommand.execute();
+        } else if (clearCommand != null) {
+            return clearCommand.execute();
+        } else if (deleteCommand != null) {
+            return deleteCommand.execute();
+        } else if (exitCommand != null) {
+            return exitCommand.execute();
+        } else if (findCommand != null) {
+            return findCommand.execute();
+        } else if (helpCommand != null) {
+            return helpCommand.execute();
+        } else if (incorrectCommand != null) {
+            return incorrectCommand.execute();
+        } else if (listCommand != null) {
+            return listCommand.execute();
+        } else if (viewAllCommand != null) {
+            return viewAllCommand.execute();
+        } else if (viewCommand != null) {
+            return viewCommand.execute();
+        } else {
+            throw new UnsupportedOperationException("None of the commands are valid.");
+        }
+    };
+
+    public Command getCommand() {
```

Comment:
I think this is still taking advantage of polymorphism because we are referring to a subtype object via its supertype.

Reply:
Hmm, true. Guess I'll have to make it looks even uglier :P

Reply:
Wait, nevermind, I take that back :-P The issue only talks about polymorphic behavior anyway, so I guess if you don't call any polymorphic methods everything is fine.